src/main/java/org/osc/manager/rest/server/api/DomainApis.java

```diff
@@ -40,19 +50,55 @@
 public class DomainApis {
 
     private static final Logger logger = Logger.getLogger(DomainApis.class);
+    private EntityManager em;
+    private TransactionControl txControl;
+
+    /**
+     * init the DB
+     *
+     *
+     */
+    public void init(EntityManager em, TransactionControl txControl) throws Exception {
+        this.em = em;
+        this.txControl = txControl;
+    }
 
     /**
      * Creates the Policy for a given domain
      *
      * @return policy
      */
-    @Path("/{domainid}")
+    @Path("/{domainid}/policies")
     @POST
     public PolicyEntity createPolicy(@PathParam("domainid") Long domainid, PolicyEntity entity) {
 
-        logger.info("Creating Policy Entity...");
-        //TODO Sudhir: Add db calls here
-        return null;
+        logger.info("Creating Policy Entity...:" + entity.getName());
+
+        return this.txControl.required(new Callable<PolicyEntity>() {
+
+            @Override
+            public PolicyEntity call() throws Exception {
+
+                CriteriaBuilder cb = em.getCriteriaBuilder();
+
+                CriteriaQuery<DomainEntity> q = cb.createQuery(DomainEntity.class);
+                Root<DomainEntity> r = q.from(DomainEntity.class);
+                q.select(r).where(cb.and(cb.equal(r.get("Id"), domainid)));
+
+                List<DomainEntity> result = em.createQuery(q).getResultList();
+                if (result.isEmpty() == false) {
+                    PolicyEntity policy = new PolicyEntity();
+                    policy.setName(entity.getName());
+                    policy.setDomain(result.get(0));
+                    policy.setParent(result.get(0).getapplianceManagerConnector());
+                    em.persist(policy);
+                    return entity;
+                }
```

Comment:
use consistent indentation

Reply:
not clear, I don't see indentation issue. the above code is within if condition.

Reply:
if {
///
} else { // dont add a break line here
///
}

Reply:
ok